packages/amplify-cli-core/src/errors/amplify-error.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Base class for all Amplify errors
+ */
+export class AmplifyError extends Error {
+  public readonly message: string;
+  public readonly resolution?: string;
+  public readonly details?: string;
+  public readonly link?: string;
+  public readonly classification?: AmplifyErrorClassification;
+
+  constructor(
+    public readonly name: AmplifyErrorType,
+    readonly options: AmplifyErrorOptions,
+  ) {
+    super(options.message);
+
+    // https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    Object.setPrototypeOf(this, AmplifyError.prototype);
+
+    this.stack ??= options.stack;
+    this.message = options.message;
+    this.details = options.details;
+    this.classification = options.classification;
+    this.resolution = 'resolution' in options ? options.resolution : undefined;
+    this.link = 'link' in options ? options.link : undefined;
+  }
+
+  toJson = (): string => {
```

Comment:
I would suggest changing this to `toObject` and have it return a serializable object. Let the errorHandler deal with the actual conversion to a string